code/game/mecha/mecha.dm

```diff
@@ -135,7 +136,7 @@
 	if(loc)
 		loc.Exited(src)
 
-	if(prob(30))
+	if(prob(30) && !noexplode)
 		explosion(get_turf(loc), 0, 0, 1, 3)
```

Comment:
Actually it's better to move explosion code out of `Destroy(..)` proc. That shouldn't be here.

Reply:
too much refactoring for now, can be done in a future mecha overhaul